localstack/services/firehose/provider.py

```diff
@@ -0,0 +1,610 @@
+    ) -> ListDeliveryStreamsOutput:
+        region = FirehoseBackend.get()
+        delivery_stream_names = []
+        for name, stream in region.delivery_streams.items():
+            delivery_stream_names.append(stream["DeliveryStreamName"])
+        return ListDeliveryStreamsOutput(
+            DeliveryStreamNames=delivery_stream_names, HasMoreDeliveryStreams=False
+        )
+
+    def put_record(
+        self,
+        context: RequestContext,
+        delivery_stream_name: DeliveryStreamName,
+        record: Record,
+    ) -> PutRecordOutput:
+        record = self._reencode_record(record)
+        return self._put_record(delivery_stream_name, record)
+
+    def put_record_batch(
+        self,
+        context: RequestContext,
+        delivery_stream_name: DeliveryStreamName,
+        records: PutRecordBatchRequestEntryList,
+    ) -> PutRecordBatchOutput:
+        records = self._reencode_records(records)
+        return PutRecordBatchOutput(
+            FailedPutCount=0, RequestResponses=self._put_records(delivery_stream_name, records)
+        )
+
+    def tag_delivery_stream(
+        self,
+        context: RequestContext,
+        delivery_stream_name: DeliveryStreamName,
+        tags: TagDeliveryStreamInputTagList,
+    ) -> TagDeliveryStreamOutput:
+        delivery_stream_description = _get_description_or_raise_not_found(
+            context, delivery_stream_name
+        )
+        FirehoseBackend.get().tags.tag_resource(
+            delivery_stream_description["DeliveryStreamARN"], tags
+        )
+        return ListTagsForDeliveryStreamOutput()
+
+    def list_tags_for_delivery_stream(
+        self,
+        context: RequestContext,
+        delivery_stream_name: DeliveryStreamName,
+        exclusive_start_tag_key: TagKey = None,
+        limit: ListTagsForDeliveryStreamInputLimit = None,
+    ) -> ListTagsForDeliveryStreamOutput:
+        delivery_stream_description = _get_description_or_raise_not_found(
+            context, delivery_stream_name
+        )
+        # The tagging service returns a dictionary with the given root name
+        tags = FirehoseBackend.get().tags.list_tags_for_resource(
+            arn=delivery_stream_description["DeliveryStreamARN"], root_name="root"
+        )
+        # Extract the actual list of tags for the typed response
+        tag_list: ListTagsForDeliveryStreamOutputTagList = tags["root"]
+        return ListTagsForDeliveryStreamOutput(Tags=tag_list, HasMoreTags=False)
+
+    def untag_delivery_stream(
+        self,
+        context: RequestContext,
+        delivery_stream_name: DeliveryStreamName,
+        tag_keys: TagKeyList,
+    ) -> UntagDeliveryStreamOutput:
+        delivery_stream_description = _get_description_or_raise_not_found(
+            context, delivery_stream_name
+        )
+        region = FirehoseBackend.get()
+        # The tagging service returns a dictionary with the given root name
+        region.tags.untag_resource(
```

Comment:
nit: Could be simplified to `FirehoseBackend.tags.untag_resource(..)`, given that `tags` is a static class member. (in fact, we could rename it to `TAGS`, to make this a bit more clear.)